### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by AtapinDmitry**
Integration tests. added

### CODE REVIEW

**Comment by pfi79** [comment]
Very many files to change in pr.
I urge you to break the pr into smaller ones in the future.

File: hlf/sdk_chcollector.go [hunk 1]
```diff
@@ -115,10 +115,16 @@ const (
...
-	dataReady chan<- struct{}, startFrom uint64, bufSize uint,
-	connectionProfile, userName, orgName string,
+func createChCollector(
+	ctx context.Context,
+	dstChName string,
+	srcChName string,
+	dataReady chan<- struct{},
+	startFrom uint64,
+	bufSize uint,
+	connectionProfile,
```
Comment (pfi79) [inline hlf/sdk_chcollector.go:125]: type string ?

File: test/integration/chcol1/chcol1_test.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+		ordererRunners []*ginkgomon.Runner
+		redisProcess   ifrit.Process
+		redisDB        *runner.RedisDB
+		networkFound   *cmn.NetworkFoundation
+		robotProc      ifrit.Process
+		skiBackend     string
+		// skiRobot       string
+		peer  *nwo.Peer
+		admin *client.UserFoundation
+		// user             *client.UserFoundation
```
Comment (pfi79) [inline test/integration/chcol1/chcol1_test.go:86]: maybe comments delete

File: test/integration/chcol1/chcol1_test.go [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+		robotProc = ifrit.Invoke(robotRunner)
+		Eventually(robotProc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+	AfterEach(func() {
+		By("stop robot")
+		if robotProc != nil {
+			robotProc.Signal(syscall.SIGTERM)
+			Eventually(robotProc.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+	})
```
Comment (pfi79) [inline test/integration/chcol1/chcol1_test.go:191]: where is "stop redis"?

File: test/integration/chcol2/chcol2_test.go [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+		robotProc = ifrit.Invoke(robotRunner)
+		Eventually(robotProc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+	AfterEach(func() {
+		By("stop robot")
+		if robotProc != nil {
+			robotProc.Signal(syscall.SIGTERM)
+			Eventually(robotProc.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+	})
```
Comment (pfi79) [inline test/integration/chcol2/chcol2_test.go:192]: where is "stop redis"?

File: test/integration/chcol3/chcol3_test.go [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+		robotProc = ifrit.Invoke(robotRunner)
+		Eventually(robotProc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+	AfterEach(func() {
+		By("stop robot")
+		if robotProc != nil {
+			robotProc.Signal(syscall.SIGTERM)
+			Eventually(robotProc.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+	})
```
Comment (pfi79) [inline test/integration/chcol3/chcol3_test.go:191]: where is "stop redis"?

File: test/integration/chexec/chexec_test.go [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+		robotProc = ifrit.Invoke(robotRunner)
+		Eventually(robotProc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+	AfterEach(func() {
+		By("stop robot")
+		if robotProc != nil {
+			robotProc.Signal(syscall.SIGTERM)
+			Eventually(robotProc.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+	})
```
Comment (pfi79) [inline test/integration/chexec/chexec_test.go:191]: where is "stop redis"?

File: test/integration/storage/storage_test.go [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+		robotProc = ifrit.Invoke(robotRunner)
+		Eventually(robotProc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+	AfterEach(func() {
+		By("stop robot")
+		if robotProc != nil {
+			robotProc.Signal(syscall.SIGTERM)
+			Eventually(robotProc.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+	})
```
Comment (pfi79) [inline test/integration/storage/storage_test.go:198]: where is "stop redis"?

File: test/integration/swap/swap_test.go [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+	"github.com/hyperledger/fabric/integration/nwo"
+	"github.com/hyperledger/fabric/integration/nwo/fabricconfig"
+	runnerFbk "github.com/hyperledger/fabric/integration/nwo/runner"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/tedsuo/ifrit"
+	ginkgomon "github.com/tedsuo/ifrit/ginkgomon_v2"
+)
+
```
Comment (pfi79) [inline test/integration/swap/swap_test.go:26]: exec go imports

File: test/integration/swap/swap_test.go [hunk 9]
```diff
@@ -0,0 +1,364 @@
...
+		robotProc = ifrit.Invoke(robotRunner)
+		Eventually(robotProc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+	AfterEach(func() {
+		By("stop robot")
+		if robotProc != nil {
+			robotProc.Signal(syscall.SIGTERM)
+			Eventually(robotProc.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+	})
```
Comment (pfi79) [inline test/integration/swap/swap_test.go:211]: where is "stop redis"?

File: test/integration/swap/swap_test.go [hunk 10]
```diff
@@ -0,0 +1,364 @@
...
+			client.Query(network, peer, cmn.ChannelCC, cmn.ChannelCC,
+				fabricnetwork.CheckResult(fabricnetwork.CheckBalance(buyAmount), nil),
+				fnBalanceOf, user.AddressBase58Check)
+
+			By("check allowed balance 3")
+			client.Query(network, peer, cmn.ChannelCC, cmn.ChannelCC,
+				fabricnetwork.CheckResult(fabricnetwork.CheckBalance(zeroAmount), nil),
+				fnAllowedBalanceOf, user.AddressBase58Check, ccFiatUpper)
+		*/
+	})
```
Comment (pfi79) [inline test/integration/swap/swap_test.go:307]: add comments - why this comment

File: hlf/sdk_chcollector.go (hunk 1 above)
Comment (AtapinDmitry) [reply hlf/sdk_chcollector.go]: fixed

File: test/integration/swap/swap_test.go (hunk 10 above)
Comment (AtapinDmitry) [reply test/integration/swap/swap_test.go]: deleted

**Review by pfi79 — APPROVED** [review]

**Pull request merged**